src/interface/ads1x15.py

```diff
@@ -0,0 +1,284 @@
+# This library
+
+import time
+import smbus
+import Adafruit_ADS1x15 #Install from here: https://github.com/adafruit/Adafruit_Python_ADS1x15
+
+ # For wiring, see https://learn.adafruit.com/raspberry-pi-analog-to-digital-converters/ads1015-slash-ads1115
+# 			  and https://cdn-learn.adafruit.com/downloads/pdf/adafruit-4-channel-adc-breakouts.pdf
+
+
+ # Resistor Divider Factor(s) Note: For a typical lipo sensing divider, R1 is the larger resistance.  http://www.ohmslawcalculator.com/voltage-divider-calculator 
+# See below for the circuit diagram
+#
+#    VBAT ------R1
+#               |  
+#               |--ADS pin
+#				|				
+#	 GND  ------R2
+#
+
+
+ # The ADS1x15 has adjustable gain. The maximum voltages at each gain are shown below.
+# The voltage at the ADS pin should not exceed the programmed gain limit below. Exceeding VDD will break the unit. 
+
+GAIN_FACTORS = {
+	2/3 : 6.144, #note, 2/3 gets cast to 0, so sending 0 in for this gain factor also works
+	1 : 4.096,
+	2 : 2.048,
+	4 : 1.024,
+	8 : 0.512,
+	16 : 0.256,
+}
+
+
+VALID_HARDWARE_NAMES = {
+	"ADS1115", 		#for the 16 bit 4-channel version
+	"ADS1015",		#for the 12 bit 4-channel version
+}
+
+VALID_I2C_ADDRESSES = { 
+	0x49, # ADR -> GND
+	0x48, # ADR -> VDD , default if ADR pin is not connected
+	0x4A, # ADR -> SDA
+	0x4B, # ADR -> SCL
+}
+
+VALID_CONNECTED_CHANNELS = {
+	0,
+    1,
+ 	2,
+ 	3,	
+}
+
+ # The code below is for scanning the i2c bus for ADS1X15 sensors
+bus = smbus.SMBus(1) # 1 indicates /dev/i2c-1
+
+
+
+
+class ADS1X15():
+	def __init__(self,
+		hardware_name = "ADS1115",		# Either ADS1115 or ADS1015
+		connected_channels = [0,], 		# Which connected channels to use, pick any from 0,1,2,3
```

Comment:
I feel the config will be better served by having an array of objects, rather than an object with array properties. It would completely remove the need to check all the arrays are of the same length.